packages/hydrogen/src/newsletter-subscribe/newsletterSubscribeHandler.ts

```diff
@@ -0,0 +1,64 @@
+import type {Storefront, StorefrontApiErrors} from '../storefront';
+import type {CustomerUserError} from '@shopify/hydrogen-react/storefront-api-types';
+export interface NewsletterSubscribeHandlerResponse {
+  isSuccessful: boolean;
+  simplifyError?: string;
+  userErrors?: CustomerUserError[];
+  apiErrors?: StorefrontApiErrors;
+}
+
+export async function newsletterSubscribeHandler(
```

Comment:
instead of exporting a single handler. Make an `createShoifyHandler` and use it in the context instead

```
// in server.ts
const shopify = createShoifyHandler(...);

....

getLoadContext: (): AppLoadContext => ({
   session,
   shopify,
   env,
   waitUntil
})
```